File: 은지/LV2/문자열 압축.js
function solution(s) {
  const n = s.length
  var answer = n
  for (let x=1; x<n; x++) {
    var exam = ''
    var pre = ''
    var cnt = 0
    for (let i=0; i<n; i+=x) {
      if (s.slice(i, i+x) !== pre) {
        if (cnt > 1) {
          exam += String(cnt) + pre
        } else {
          exam += pre
        }
        pre = s.slice(i, i+x)
        cnt = 1
        if (i >= n-x) {
          exam += pre
        }
      } else {
        if (i >= n-x) {
          exam += String(cnt+1) + s.slice(i, i+x)
        }
        cnt += 1
      }
    }
    if (answer > exam.length) {
      answer = exam.length
    }
  }
  return answer;
}

s = "abcabcdede"
console.log(solution(s))

File: 은지/LV2/문자열 압축.py
def solution(s):
    n = len(s)
    answer = n
    # 문자열 자르기
    for x in range(1, n):
        exam = ''
        pre = ''
        cnt = 0
        for i in range(0, n, x):
            # 과거 문자열과 현재 문자열이 다른 경우
            if s[i:i+x] != pre:
                if cnt > 1:
                    exam += str(cnt) + pre
                else:
                    exam += pre
                pre = s[i:i+x]
                # 개수 초기화
                cnt = 1
                # 마지막 문자열
                if i >= n-x:
                    exam += pre
            # 과거 문자열과 현재 문자열이 같은 경우
            else:
                # 마지막 문자열
                if i >= n-x:
                    exam += str(cnt+1) + s[i:i+x]
                cnt += 1
        # 최소 문자열 탐색
        if answer > len(exam):
            answer = len(exam)
    return answer

s = "abcabcdede"
print(solution(s))